gateway-release/home/conf/gateway-site.xml

```diff
@@ -18,6 +18,11 @@ limitations under the License.
 -->
 <configuration>
 
+    <property>
+        <name>gateway.services</name>
```

Comment:
I had envisioned the use of distinct properties, rather than a single "list" property.
Something more explicit, like:
`<property><name>gateway.service.alias</name><value>org.apache.knox.gateway.services.security.impl.RemoteAliasService</value></property>
<property><name>gateway.service.tokenstate</name><value>org.apache.knox.gateway.services.token.impl.JournalBasedTokenStateService</value></property>`

I think explicit class name values will make it easier for developers to create and deploy Knox customizations/extensions. Given this, I think a _list_ of such things could quickly become difficult to read.

Reply:
I do like the use of the AbstractFactory pattern here.

Reply:
> I had envisioned the use of distinct properties, rather than a single "list" property.
> Something more explicit, like:
> `<property><name>gateway.service.alias</name><value>org.apache.knox.gateway.services.security.impl.RemoteAliasService</value></property> <property><name>gateway.service.tokenstate</name><value>org.apache.knox.gateway.services.token.impl.JournalBasedTokenStateService</value></property>`
> 
> I think explicit class name values will make it easier for developers to create and deploy Knox customizations/extensions. Given this, I think a _list_ of such things could quickly become difficult to read.

Thanks, @pzampino for your review comment. It makes sense and I'm in favor of your recommended approach with a small alteration: I'd build the property names as `gateway.service.$SERVICE.$PARAM` where service references the service name and param is an arbitrary parameter name that Knox can interpret later on. In this round, I'd add `impl` only but in upcoming changes, there will be others. For instance:
```
<property>
   <name>gateway.service.tokenstate.impl</name>
   <value>org.apache.knox.gateway.services.token.impl.ZookeeperTokenStateService</value>
</property>
<property>
   <name>gateway.service.tokenstate.useLocalAliasService</name>
   <value>false</value>
   <description>Indicates whether ZookeeperTokenStateService should use a local alias service when storing tokens or purely do the job in Zookeeper</description>
</property>

Any objection?

Reply:
I am strongly in favor of your suggestion. I believe it's a more flexible pattern than list-value properties. My examples had originally included `gateway.service.$SERVICE.impl` as property names, but I modified them for simplicity, since it was only an example, and not the primary point of my comment.

Reply:
Fixed them all. Thanks again for the review, @pzampino !